### PR TITLE
Integrate Contact Details API with FT

### PR DESCRIPTION
**Pull request by LBHCallumM**
## Link to JIRA ticket

[Ticket](https://www.pivotaltracker.com/story/show/180886400)

## Describe this PR

I have added an integration with ContactDetailsAPI, with a FeatureToggle.

I have also included some additional information in the response, which optionally can be used:
- TenureType e.g. "HouseholdMember || Tenant". Indicates who the contact is.
- Phone Number type e.g. "mobile || home || other"
- Phone Number description. Free text that is used to summarise the phone number.

#### _Checklist_

- [ ] Added end-to-end (i.e. integration) tests where necessary e.g to test complete functionality of a newly added feature
- [ ] Added tests to cover all new production code
- [ ] Added comments to the README or updated relevant documentation _(add link to documentation)_, where necessary.
- [ ] Checked all code for possible refactoring
- [ ] Code pipeline builds correctly

### CODE REVIEW

File: RepairsApi.Tests/E2ETests/GetPropertyE2ETests.cs [hunk 1]
```diff
@@ -133,6 +135,38 @@ public void GetPropertyWithAlerts(int propertyAlertCount, int personAlertCount,
...
+            var response = client.ExecuteRequest<PropertyResponse>("dummy", new Uri($"/api/v2/properties/{propertyReference}", UriKind.Relative)).Result;
+
+            // Assert
+            response.IsSuccess.Should().BeTrue();
+            response.Status.Should().Be(HttpStatusCode.OK);
+
+            response.Content.ContactDetails.Should().NotBeNull();
+            response.Content.ContactDetails.Should().HaveCount(contactDetailsResponse.Results.Count());
+        }
+
```
Comment (Asura5) [inline RepairsApi.Tests/E2ETests/GetPropertyE2ETests.cs:169]: When no contact details are found, return 404?

File: RepairsApi/V2/Domain/ResidentPhoneNumber.cs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+namespace RepairsApi.V2.Domain
+{
+    public class ResidentPhoneNumber
+    {
+        public string Type { get; set; }
```
Comment (Asura5) [inline RepairsApi/V2/Domain/ResidentPhoneNumber.cs:5]: Is this defined strictly as Type anywhere? My thoughts are that having this as Type will be confusing for developers. who are expecting Type to be a .NET CLR Type. Could be PhoneNumberType?

Comment (Asura5) [reply RepairsApi/V2/Domain/ResidentPhoneNumber.cs:5]: Any could it be an Enum?

File: RepairsApi.Tests/E2ETests/GetPropertyE2ETests.cs (hunk 1 above)
Comment (LBHCallumM) [reply RepairsApi.Tests/E2ETests/GetPropertyE2ETests.cs:169]: The way the API works, I have to call the api once per contact. If you have a look at ContactDetailsGateway, I return an empty list if no contactDetails are found.

File: RepairsApi/V2/Gateways/ContactDetailsGateway.cs [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+            var taskList = householdMembers.Select(x => GetContactDetails(x)).ToList();
+
+            // wait for all tasks to complete
+            await Task.WhenAll(taskList);
+
+            // return output of all tasks
+            return taskList.Select(x => x.Result).Where(x => x != null);
+        }
+
+        private async Task<ResidentContactDetails> GetContactDetails(HouseholdMember householdMember)
```
Comment (Asura5) [inline RepairsApi/V2/Gateways/ContactDetailsGateway.cs:45]: Naming convention: missing Async

File: RepairsApi/V2/Gateways/ContactDetailsGateway.cs [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+            _logger = logger;
+        }
+
+        public async Task<IEnumerable<ResidentContactDetails>> GetContactDetailsAsync(IEnumerable<HouseholdMember> householdMembers)
+        {
+            var contactDetails = new List<ResidentContactDetails>();
+
+            if (householdMembers == null || !householdMembers.Any())
+            {
+                // return empty list
```
Comment (Asura5) [inline RepairsApi/V2/Gateways/ContactDetailsGateway.cs:31]: Should be a warning logged on this almost-unhappy path

File: RepairsApi.Tests/E2ETests/GetPropertyE2ETests.cs (hunk 1 above)
Comment (LBHCallumM) [reply RepairsApi.Tests/E2ETests/GetPropertyE2ETests.cs:169]: I do have a test in `ContactDetailsGatewayTests.GetContactDetailsAsync_WhenOneContactHasNoContactDetails_NullIsNotAddedToTheResponse` for when one contact returns 404.

**Review by Asura5 — APPROVED** [review]

**Pull request merged**